File: python/snippets/recipes/event_type_in_or.py
"""Filter records where event_type is either 'login' or 'logout'.

FlyQL query: ``event_type = 'login' or event_type = 'logout'``

Note: text values like ``'login'`` MUST be quoted. Without quotes, FlyQL would
treat ``login`` as a column reference. See ``syntax/values`` for details.
"""

from flyql import parse
from flyql.generators.clickhouse.column import Column
from flyql.generators.clickhouse.generator import to_sql_where

result = parse("event_type = 'login' or event_type = 'logout'")

columns = {
    "event_type": Column("event_type", "String"),
}

sql = to_sql_where(result.root, columns)
print(sql)

File: javascript/packages/flyql/snippets/recipes/event-type-in-or.mjs
// Filter records where event_type is either 'login' or 'logout'.
//
// FlyQL query: event_type = 'login' or event_type = 'logout'
//
// Text values like 'login' MUST be quoted; without quotes, FlyQL would treat
// `login` as a column reference. See syntax/values for details.

import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse("event_type = 'login' or event_type = 'logout'")

const columns = {
    event_type: newColumn({ name: 'event_type', type: 'String' }),
}

const sql = generateWhere(result.root, columns)
console.log(sql)
